### ISSUE TITLE
How to access Electron logs

### ISSUE THREAD

**Issue by emorikawa** [kevinsawicki/spectron#5]
I can't seem to access any logs from neither the browser process nor the render processes. Any ideas on how to do this?

Selenium has a `/session/:sessionId/log` and `/session/:sessionId/log/types` endpoint that I'm successfully able to access via the `webdriver` `client.log()` and `client.logTypes()` APIs. `logTypes` resolves to an array with `['browser', 'driver']`.

Unfortunately, the value of the logs coming out of the `browser` type is always empty. I can see that the app boots, and I can open the inspector console and see logs and enter more `console.log` commands, but nothing ever comes back when you request the logs via `client.log`.

```javascript
this.app.start().then(()=> {
  this.app.client.log("browser").then(({value}) => {
     console.log(value) // Always an empty []
  })
})
```

I've also tried putting that log polling function on an interval thinking that it wouldn't return anything until the app fully booted. It never returns any data.

On the flip side, polling `app.client.log("driver")` returns all of the Selenium driver log data properly.

**Comment by kevinsawicki** [comment]
I was able to get the render process logs via `app.client.log("browser")`

I just pushed a commit to master with a test for it and a special API called `getRenderProcessLogs()`.

https://github.com/kevinsawicki/spectron/commit/f5193cb90d1e2a4ad591a5f1e6185f8097d19905

Have you tried waiting on `waitUntilWindowLoaded` before retrieving the logs?

**Comment by kevinsawicki** [comment]
I've added a `app.client.getMainProcessLogs()` helper that gets the main process logs and clears them.

I published `0.35.1` with this change, can you give it a try and see if it gets you the information you are looking for?

**Comment by kevinsawicki** [comment]
I've published `0.35.1` which contains two helpers, `getMainProcessLogs()` and `getRenderProcessLogs()` that I believe address what you were looking for.

Please let me know if they are missing anything, going to close out this issue for now, thanks for reporting this.
